ui/revenuecatui/src/main/kotlin/com/revenuecat/purchases/ui/revenuecatui/templates/template2/Template2.kt

```diff
@@ -0,0 +1,160 @@
+        Row(
+            modifier = Modifier
+                .fillMaxWidth()
+                .padding(
+                    start = UIConstant.defaultHorizontalPadding,
+                    end = UIConstant.defaultHorizontalPadding,
+                    bottom = UIConstant.defaultButtonVerticalSpacing,
+                ),
+            horizontalArrangement = Arrangement.Center,
+        ) {
+            RestorePurchasesButton(viewModel)
+        }
+    }
+}
+
+@Composable
+private fun Template2MainContent(state: PaywallViewState.Template2, viewModel: PaywallViewModel) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .verticalScroll(rememberScrollState())
+            .padding(horizontal = UIConstant.defaultHorizontalPadding),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.spacedBy(UIConstant.defaultButtonVerticalSpacing, Alignment.CenterVertically),
+    ) {
+        // TODO-PAYWALLS: Replace with correct image
+        val drawable = LocalContext.current.packageManager.getApplicationIcon(LocalContext.current.packageName)
+        Image(drawable.toBitmap(config = Bitmap.Config.ARGB_8888).asImageBitmap(), contentDescription = "")
+
+        val localizedConfig = state.paywallData.localizedConfig()
+        Text(
+            style = MaterialTheme.typography.displaySmall,
+            fontWeight = FontWeight.Black,
+            textAlign = TextAlign.Center,
+            text = localizedConfig?.title ?: "",
+            color = state.paywallData.getColors().text1Color,
+        )
+        Text(
+            style = MaterialTheme.typography.bodyLarge,
+            fontWeight = FontWeight.Bold,
+            textAlign = TextAlign.Center,
+            text = localizedConfig?.subtitle ?: "",
+            color = state.paywallData.getColors().text1Color,
+        )
+        state.packages.forEach { aPackage ->
+            SelectPackageButton(state, aPackage, viewModel)
+        }
+    }
+}
+
+@Composable
+private fun PurchaseButton(state: PaywallViewState.Template2, viewModel: PaywallViewModel) {
+    Column(
+        modifier = Modifier
+            .fillMaxWidth()
+            .padding(horizontal = UIConstant.defaultHorizontalPadding),
+        horizontalAlignment = Alignment.CenterHorizontally,
+        verticalArrangement = Arrangement.Center,
+    ) {
+        val activity = LocalContext.current.getActivity() ?: error("Error finding activity")
+        val colors = state.paywallData.getColors()
+        Button(
+            modifier = Modifier.fillMaxWidth(),
+            onClick = { viewModel.purchaseSelectedPackage(activity) },
+            colors = ButtonDefaults.buttonColors(
+                containerColor = colors.callToActionBackgroundColor,
+                contentColor = colors.callToActionForegroundColor,
+            ),
+        ) {
+            Text(text = state.paywallData.localizedConfig()?.callToAction ?: "")
```

Comment:
For the future: this will need to display `callToActionWithOffer` if applicable.